src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ToolStripContainerDesignerTests.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.ComponentModel;
+using System.ComponentModel.Design;
+using System.Reflection;
+using Moq;
+
+namespace System.Windows.Forms.Design.Tests;
+
+public class ToolStripContainerDesignerTests : IDisposable
+{
+    private readonly Mock<IDesignerHost> _mockDesignerHost;
+    private readonly Mock<ISelectionService> _mockSelectionService;
+    private readonly ToolStripContainerDesigner _designer;
+    private readonly ToolStripContainer _toolStripContainer;
+
+    public ToolStripContainerDesignerTests()
+    {
+        _mockDesignerHost = new();
+        _mockSelectionService = new();
+        _toolStripContainer = new();
+        InitializeMocks();
+
+        _designer = new();
+        _designer.Initialize(_toolStripContainer);
+    }
+
+    private void InitializeMocks()
+    {
+        Mock<ISite> mockSite = new();
+        mockSite.Setup(s => s.GetService(typeof(IDesignerHost))).Returns(_mockDesignerHost.Object);
+        mockSite.Setup(s => s.GetService(typeof(ISelectionService))).Returns(_mockSelectionService.Object);
+        _toolStripContainer.Site = mockSite.Object;
+    }
```

Comment:
Let's make this a local function since it is only being used in the constructor.
```suggestion
        _designer.Initialize(_toolStripContainer);

        void InitializeMocks()
        {
            Mock<ISite> mockSite = new();
            mockSite.Setup(s => s.GetService(typeof(IDesignerHost))).Returns(_mockDesignerHost.Object);
            mockSite.Setup(s => s.GetService(typeof(ISelectionService))).Returns(_mockSelectionService.Object);
            _toolStripContainer.Site = mockSite.Object;   
        }
    }
```